common/k8s/src/event_source.rs

```diff
@@ -261,198 +261,239 @@ impl K8sEventStream {
-                                let latest_event_time = latest_event_time.clone();
-                                let earliest = previous_event_logger_delete_time.clone();
-                                let ret = latest_event_time
-                                    .load()
-                                    .or_else(|| earliest.load())
-                                    .and_then(|earliest| {
-                                        let earliest = chrono::NaiveDateTime::from_timestamp(
-                                            earliest.into(),
-                                            0,
-                                        );
-                                        event.as_ref().ok().and_then(|e| {
-                                            e.last_timestamp
-                                                .as_ref()
-                                                .map(|l| earliest < l.0.naive_utc())
-                                        })
-                                    });
-                                async move { ret.unwrap_or(true) }
-                            }
-                        })
-                        .map(move |event| {
-                            match event.map(|e| {
-                                let latest_event_time = latest_event_time_w.clone();
-                                let this_event_time = e
-                                    .last_timestamp
+                            })
+                            .filter_map(|e| async move {
+                                match e {
+                                    Cont::Cont => None,
+                                    Cont::Break => Some(((), ())),
+                                }
+                            });
+                        pin_mut!(stream);
+                        Ok(stream.next().await)
+                    }
+                } else {
+                    Ok(Some(((), ())))
+                }
+            }
+        };
+        try_unfold((), waiter).map(|r: Result<(), K8sEventStreamError>| match r {
+            Ok(_) => Ok(StreamElem::Waiting),
+            Err(e) => Err(e),
+        })
+    }
+
+    pub fn active_stream(
+        client: Arc<Client>,
+        latest_event_time: Arc<AtomicCell<Option<NonZeroI64>>>,
+        previous_event_logger_delete_time: Arc<AtomicCell<Option<NonZeroI64>>>,
+    ) -> impl Stream<Item = Result<StreamElem<LineBuilder>, K8sEventStreamError>> {
+        let events: Api<Event> = Api::all(client.as_ref().clone());
+        let params = ListParams::default();
+
+        let latest_event_time_w = latest_event_time.clone();
+        try_flatten_touched(watcher(events, params))
+            .map_err(K8sEventStreamError::WatcherError)
+            .filter({
+                move |ref event| {
+                    let latest_event_time = latest_event_time.clone();
+                    let earliest = previous_event_logger_delete_time.clone();
```

Comment:
If the pod is starting and its the oldest one, Will this be `None`? If that's the case, is that the behaviour we want?

Reply:
Yes, if there is no earliest event then we go back to the start of the event stream and log from there.

The is the equivalent of Lookback::Start for files except that rather than the lookback happening on pod restart it should only happen if the entire agent install is removed and readded

Reply:
gotcha 👍